src/assignment-hooks.ts

```diff
@@ -8,21 +8,22 @@ export interface IAssignmentHooks {
   /**
    * Invoked before a subject is assigned to an experiment variation.
    *
-   * @param experimentKey key of the experiment being assigned
+   * @param flagKey key of the feature flag being used for assignment
    * @param subject id of subject being assigned
    * @returns variation to override for the given subject. If null is returned,
    * then the subject will be assigned with the default assignment logic.
    * @public
    */
-  onPreAssignment(experimentKey: string, subject: string): EppoValue | null;
+  onPreAssignment(flagKey: string, subject: string): EppoValue | null;
 
   /**
    * Invoked after a subject is assigned. Useful for any post assignment logic needed which is specific
-   * to an experiment/flag. Do not use this for logging assignments - use IAssignmentLogger instead.
-   * @param experimentKey key of the experiment being assigned
+   * to a flag or allocation. Do not use this for logging assignments - use IAssignmentLogger instead.
+   * @param flagKey key of the feature flag being used for assignment
    * @param subject id of subject being assigned
    * @param variation the assigned variation
+   * @param allocationKey key of the allocation being used for assignment
    * @public
    */
-  onPostAssignment(experimentKey: string, subject: string, variation: EppoValue | null): void;
+  onPostAssignment(flagKey: string, subject: string, variation: EppoValue | null, allocationKey?: string | null): void;
```

Comment:
This is, I think, the one decision point this PR where there was some ambiguity.

After assignment, we could either:
1. keep the original `experimentKey` argument and pass in the experiment key  \` ${flagKey}-${allocationKey}\` instead of the just the `flagKey`
2. or change the first arg to `flagKey` argument and pass in the allocation key as an optional argument

I went with option 2. here because 

* It keeps all the original functionality of the `onPostAssignment` function (i.e. not changing the format of the string that gets passed into the function's first argument) and consequently doesn't require an immediate code change
* It also enables custom allocation-level post assignment functionality if necessary, but would require the user to be aware of `allocationKeys` to use this correctly

If I went with 1, the `onPostAssignment` code (customer-end) would need to be updated to handle the new key format, which works against out goal of making this transition to logging `allocationId`s as friction-free as possible

Reply:
Ah the options idea is a cool one, but agreed that it's a bit messy